clang/lib/Serialization/ASTWriterDecl.cpp

```diff
@@ -650,17 +642,16 @@ void ASTDeclWriter::VisitFunctionDecl(FunctionDecl *D) {
     DependentFunctionTemplateSpecializationInfo *
       DFTSInfo = D->getDependentSpecializationInfo();
 
-    // Templates.
-    Record.push_back(DFTSInfo->getNumTemplates());
-    for (int i=0, e = DFTSInfo->getNumTemplates(); i != e; ++i)
-      Record.AddDeclRef(DFTSInfo->getTemplate(i));
+    // Candidates.
+    Record.push_back(DFTSInfo->getCandidates().size());
```

Comment:
`getNumTemplates()` returns `unsigned` while `size()` AFAICT returns `size_t` which is not the same type. Is this ok?

Reply:
`DependentFunctionTemplateSpecializationInfo` stores the number of candidates in an `unsigned` member, so it should be fine.